packages/ethereum/contracts/proxy/HoprStakingProxyForNetworkRegistry.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+import '@openzeppelin/contracts/access/Ownable.sol';
+import '../IHoprNetworkRegistryRequirement.sol';
+
+contract IStake {
+  function stakedHoprTokens(address _account) public view returns (uint256) {}
+
+  function isNftTypeAndRankRedeemed3(
+    uint256 nftTypeIndex,
+    uint256 boostNumerator,
+    address hodler
+  ) external view returns (bool) {}
+}
+
+contract HoprStakingProxyForNetworkRegistry is IHoprNetworkRegistryRequirement, Ownable {
+  struct NftTypeAndRank {
+    uint256 nftType;
+    uint256 nftRank;
+  }
+
+  IStake public immutable STAKE_CONTRACT;
+  NftTypeAndRank[] public eligibleNftTypeAndRank;
+  uint256 public stakeThreshold;
+
+  event AddedNftTypeAndRank(uint256 indexed nftType, uint256 indexed nftRank);
+  event RemovedNftTypeAndRank(uint256 indexed nftType, uint256 indexed nftRank);
+  event UpdatedThreshold(uint256 indexed threshold);
+
+  constructor(
+    address stakeContract,
+    address newOwner,
+    uint256 minStake
+  ) {
+    STAKE_CONTRACT = IStake(stakeContract);
+    stakeThreshold = minStake;
+    emit UpdatedThreshold(stakeThreshold);
+    transferOwnership(newOwner);
+  }
+
+  /**
+   * @dev Checks if the msg.sender has staked any NFT of eligibleNftTypeAndRank and
+   * if staked token amount is above `threshold` of staked HOPR tokens
+   * @param account staker address that has a hopr nodes running
+   */
+  function isRequirementFulfilled(address account) external view returns (bool) {
+    // for self-claiming accounts, check against the current criteria
+    uint256 amount = STAKE_CONTRACT.stakedHoprTokens(account);
+    if (amount < stakeThreshold) {
+      // threshold does not meet
+      return false;
+    }
+
+    for (uint256 i = 0; i < eligibleNftTypeAndRank.length; i++) {
+      NftTypeAndRank memory eligible = eligibleNftTypeAndRank[i];
+      if (STAKE_CONTRACT.isNftTypeAndRankRedeemed3(eligible.nftType, eligible.nftRank, account)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
```

Comment:
if we decide to go ahead with `dev NFTs`, would this would be the place to update the proxy?

Reply:
If we want to issue "dev NFTs", owner should call `ownerBatchAddNftTypeAndRank` or `ownerAddNftTypeAndRank` to add the dev NFT in the "HoprStakingProxyForNetworkRegistry" proxy contract.
If we don't want to use "dev NFTs", owner can add dev addresses with `ownerRegister` in "HoprNetworkRegistry" contract.

Reply:
but then wouldn't that require devs to have staked? I don't think this function will work out of the box

Reply:
No need to stake. `specialeNftTypeAndRank` checks it